tests/unit/models/Component.spec.js

```diff
@@ -539,5 +567,787 @@ describe('Test class: Component', () => {
+            type: 'String',
+          });
+
+          const attribute = new ComponentAttribute({
+            name: 'attribute-string',
+            value: 1,
+            type: 'String',
+            definition: attributeDefinitionType,
+          });
+
+          expect(new Component({
+            attributes: [attribute],
+            definition: {
+              definedAttributes: [attributeDefinitionType],
+            },
+          }).hasError()).toEqual(true);
+        });
+      });
+
+      describe('Test Boolean attribute', () => {
+        it('Should return false if value type is a boolean', () => {
+          const attributeDefinition = new ComponentAttributeDefinition({
+            name: 'attribute-boolean',
+            type: 'Boolean',
+          });
+
+          const attribute = new ComponentAttribute({
+            name: 'attribute-boolean',
+            value: false,
+            type: 'Boolean',
+            definition: attributeDefinition,
+          });
+
+          expect(new Component({
+            attributes: [attribute],
+            definition: { definedAttributes: [attributeDefinition] },
+          }).hasError()).toEqual(false);
+        });
+
+        it('Should return true if value type isn\'t boolean', () => {
+          const attributeDefinition = new ComponentAttributeDefinition({
+            name: 'attribute-boolean',
+            type: 'Boolean',
+          });
+
+          const attribute = new ComponentAttribute({
+            name: 'attribute-boolean',
+            value: 'string',
+            type: 'Boolean',
+            definition: attributeDefinition,
+          });
+
+          expect(new Component({
+            attributes: [attribute],
+            definition: { definedAttributes: [attributeDefinition] },
+          }).hasError()).toEqual(true);
+        });
+      });
+
+      describe('Test Number attribute', () => {
+        const attributeDefinition = new ComponentAttributeDefinition({
+          name: 'attribute-number',
+          type: 'Number',
+          rules: {
+            min: 5,
+            max: 10,
+          },
+        });
+
+        it('Should return false if the "value" rule includes the attribute\'s values', () => {
+          const attributeDefinitionValue = new ComponentAttributeDefinition({
+            name: 'attribute-number',
+            type: 'Number',
+            rules: { values: [0, 1] },
+          });
+
+          const attribute = new ComponentAttribute({
+            name: 'attribute-number',
+            value: 1,
+            type: 'Number',
+            definition: attributeDefinitionValue,
+          });
+
+          expect(new Component({
+            attributes: [attribute],
+            definition: {
+              definedAttributes: [attributeDefinitionValue],
+            },
+          }).hasError()).toEqual(false);
+        });
+
+        it(
+          'Should return true if the "value" rule doesn\'t includes the attribute\'s values',
```

Comment:
'Should return true if the "values" rule doesn\'t includes the attribute\'s value',

Reply:
nope l.806